dom/src/makeDOMDriver.ts

```diff
@@ -1,75 +1,97 @@
 import {StreamAdapter} from '@cycle/base';
-import {init} from 'snabbdom';
+import {init as initSnabbdom, PatchFunction} from 'snabbdom';
 import xs, {Stream} from 'xstream';
 import {DOMSource} from './DOMSource';
 import {MainDOMSource} from './MainDOMSource';
 import {VNode} from './interfaces';
 import {VNodeWrapper} from './VNodeWrapper';
 import {getElement} from './utils';
 import defaultModules from './modules';
-import {IsolateModule} from './isolateModule';
+import {IsolateModule} from './IsolateModule';
 import {makeTransposeVNode} from './transposition';
 import {EventDelegator} from './EventDelegator';
-import xsSA from '@cycle/xstream-adapter';
+import xsAdapter from '@cycle/xstream-adapter';
 let MapPolyfill: typeof Map = require('es6-map');
 
-function makeDOMDriverInputGuard(modules: any) {
+export {makeDOMDriver}
+
+export interface DOMDriverOptions {
+  modules?: Array<any>;
+  transposition?: boolean;
+}
+
+function makeDOMDriver(
+  container: string | Element,
+  options: {transposition?: boolean, modules?: Array<any>} = {}
+): Function {
+  const {transposition = false, modules = defaultModules} = options;
+
+  makeDOMDriverModulesGuard(modules);
+
+  const isolateModule = new IsolateModule(new MapPolyfill<string, Element>());
+  const eventDelegators: Map<string, EventDelegator> =
+    new MapPolyfill<string, EventDelegator>();
+
+  const patch: PatchFunction = initSnabbdom([isolateModule.createModule()]
+                                 .concat(modules));
+
+  const rootElement: Element = getElement(container);
+  const vnodeWrapper = new VNodeWrapper(rootElement);
+
+  function _DOMDriver(vNode$: Stream<VNode>,
+                      runStreamAdapter: StreamAdapter,
+                      driverKey: string): DOMSource {
+    domDriverVNodeStreamGuard(vNode$);
+
+    const transposeVNode: (vNode: VNode) => xs<VNode> =
```

Comment:
With TypeScript's type inference in place doing it's job well, I find it unnecessary to give the type for these, it adds some visual noise and makes it harder to read. `(vnode: VNode) => xs<VNode>` sort of looks like an array function as an implementation, but it's just a type signature. I find it more readable without the signature, and it's still easily verifiable in TypeScript and in some editors if you hover the mouse over the const transposeVNode.